climada/util/yearsets.py

```diff
@@ -0,0 +1,247 @@
+"""
+This file is part of CLIMADA.
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU Lesser General Public License as published by the Free
+Software Foundation, version 3.
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+You should have received a copy of the GNU Lesser General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+---
+Define functions to handle impact_yearsets
+"""
+
+import copy
+import logging
+import numpy as np
+from numpy.random import default_rng
+
+import climada.util.dates_times as u_dt
+
+LOGGER = logging.getLogger(__name__)
+
+def impact_yearset(event_impacts, sampled_years=None, sampling_dict=None, correction_fac=True):
+
+    """Create an annual_impacts object containing a probabilistic impact for each year
```

Comment:
I think it could be nice to add a small sentence explaining how this is different to the expected annual impact, I guess the difference is already explained in the "impact for **each** year" but it still took me while to understand